phoenix-core/src/main/java/org/apache/phoenix/compile/CreateTableCompiler.java

```diff
@@ -160,9 +170,23 @@ public MutationPlan compile(CreateTableStatement create) throws SQLException {
                     viewColumnConstantsToBe = new byte[nColumns][];
                     ViewWhereExpressionVisitor visitor = new ViewWhereExpressionVisitor(parentToBe, viewColumnConstantsToBe);
                     where.accept(visitor);
+
+                    viewTypeToBe = visitor.isUpdatable() ? ViewType.UPDATABLE : ViewType.READ_ONLY;
+                    if (viewTypeToBe == ViewType.UPDATABLE) {
+                        ViewWhereExpressionValidatorVisitor validatorVisitor =
+                                new ViewWhereExpressionValidatorVisitor(parentToBe,
+                                        pkColumnsInWhere, nonPkColumnsInWhere);
+                        where.accept(validatorVisitor);
+                        try {
+                            viewTypeToBe = setViewTypeToBe(connection, parentToBe, pkColumnsInWhere,
+                                    nonPkColumnsInWhere);
```

Comment:
test failures are due to 
```
    @Override
    public Table getTable(byte[] tableName) throws SQLException {
        throw new UnsupportedOperationException();
    }
```

let's avoid calling `setViewTypeToBe` if we are going to use `ConnectionlessQueryServicesImpl`, no need to remove tests.

Reply:
setViewTypeToBe needs connection to SYSTEM_CHILD_LINK_TABLE, which is used to get sibling views and mark if view is updatable. How can we avoid calling it?

Reply:
I meant only for `ConnectionlessQueryServicesImpl`, not for `ConnectionQueryServicesImpl`

Reply:
but we need `setViewTypeToBe` to mark if the view is updatable, so for `ConnectionlessQueryServicesImpl` how can we make it work?